recipes/extending-kysely.md

```diff
@@ -340,7 +340,47 @@ db.insertInto('t')
 
 ## Extending using inheritance
 
-You usually don't want to do this. Because of the complex types, you'll quickly run into problems.
-Even though Kysely uses classes, Kysely is not designed from the OOP point of view. Classes are
-used because they are supported natively by Typescript. They provide private variables and a nice
-discoverable API.
+You usually don't want to do this. Because of the complex types and typescript's limitations 
+when it comes to inheritence and return types, you'll quickly run into problems.
+Even though Kysely uses classes, Kysely is not designed from the OOP point of view. 
+Classes are used because they are supported natively by Typescript. They provide private 
+variables and a nice discoverable API.
+
+## Extending using module augmentation
+
+You don't want to do this. This method can bypass Typescript's shortcomings when it comes 
```

Comment:
I think the disclaimer is enough and we can remove the first sentence. If we leave it there, this is a little bit too scary 😄

Reply:
Almost wrote that augmentation is the devil! 👹